content/en/blog/2024/prom-and-otel/index.md

```diff
@@ -0,0 +1,671 @@
+  [Kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)
+  and sends it down the metric pipeline for further processing
+- [Host Metrics Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver):
+  scrapes system metrics from hosts that make up the cluster
+
+Processing data:
+
+- [Kubernetes Attributes Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor):
+  adds Kubernetes context, thereby
+- enabling you to correlate application telemetry with your Kubernetes telemetry
+- – considered one of the most important components for monitoring Kubernetes
+- with OpenTelemetry
+
+You can also use the Kubernetes attributes processor to set custom resource
+attributes for traces, metrics, and logs using the Kubernetes labels and
+annotations you’ve added to your pods and namespaces.
+
+There are a few more Collector components you can implement to monitor
+Kubernetes, including Kubernetes-specific ones as well as general-use
+processors, such as the batch, memory limiter, and resource processors. To learn
+more, see
+[Important Components for Kubernetes](/docs/kubernetes/collector/components/).
+
+After you’ve configured the components in your Collector config file, you need
+to enable them within the [pipelines](/docs/collector/configuration/#pipelines)
+section. A data pipeline enables you to
+[collect](/docs/collector/configuration/#receivers),
+[process](/docs/collector/configuration/#processors), and route data from any
+source [to one destination or more](/docs/collector/configuration/#exporters).
+
+## Pros and cons
+
+The following are pros and cons of the setup we covered in this article.
+
+**Pros:**
+
+- Not having to maintain Prometheus as your data store, which means less
+  infrastructure overall to maintain -- especially if you go with an all-in-one
+  observability backend to ingest OTel data (traces, metrics, logs).
+- Not having to maintain the Prometheus Operator; while you would still have to
+  maintain the ServiceMonitor and PodMonitor, it’s a lot less work than keeping
+  the Operator up-to-date.
+- Allows you to end up with a full OTel solution while still obtaining your
+  Prometheus metrics
+- OTel can provide traces and logs in addition to metrics, as well as
+  correlation of these signals, thus enhancing the observability of Kubernetes
+  environments.
+- OTel provides handy tools, such as the Target Allocator and OTel Collector
+  components, to provide flexibility for configuration and deployment options.
+
+**Cons:**
+
+- Adopting and managing a new observability tool involves a steep learning curve
+  for users unfamiliar with OTel concepts, components, and workflows.
+- If you are used to using PromQL, Prometheus’ powerful query language, you may
+  have to learn how to use your backend’s query language.
```

Comment:
users can still use PromQL if they send to a prometheus-compatible DB, though this is one of many options...